torch/distributed/algorithms/ddp_comm_hooks/__init__.py

```diff
@@ -21,11 +26,30 @@ class DDPCommHookType(Enum):
         ddp_comm_hook_wrapper, comm_hook=quantization.quantization_perchannel_hook
     )
 
-# Registers the hooks of ``torch.distributed.algorithms.ddp_comm_hooks`` by taking ``comm_hook_name``
-# input as a ``str```. Also, checks whether hook is in ``DDPCommHookType``.
-def register_ddp_comm_hook(comm_hook_name, model, state=None):
+
+def register_ddp_comm_hook(
+    comm_hook_type: DDPCommHookType, model: DistributedDataParallel, state=None
+):
+    """
+        Registers the hooks of ``torch.distributed.algorithms.ddp_comm_hooks``
+        to the DDP model. User can specify the type of hook as an enum
+        ``DDPCommHookType`` type using ``comm_hook_type`` input. State input will
+        be passed to the model.
+
+        Example::
+            >>> register_ddp_comm_hook(DDPCommHookType.FP16_COMPRESS, model, state)
+    """
+    DDPCommHookType.value(model=model, state=state)
+
+
+def register_ddp_comm_hook_string_wrapper(comm_hook_name, model, state=None):
```

Comment:
Can we just have `register_ddp_comm_hook` as one way to register the hooks and get rid of the wrapper? We can update the benchmark to use `register_ddp_comm_hook` right?

Reply:
Sure.